omniqueue/src/backends/redis/mod.rs

```diff
@@ -90,6 +90,53 @@ impl RedisConnection for RedisClusterConnectionManager {
     }
 }
 
+async fn check_eviction_policy<R: RedisConnection>(pool: bb8::Pool<R>) {
+    const FAIL_MSG: &str = "Unable to verify eviction policy. Ensure `maxmemory-policy` set to `noeviction` or `volatile-*`";
+    let mut conn = match pool.get().await {
+        Ok(conn) => conn,
+        Err(e) => {
+            tracing::debug!("{e}");
```

Comment:
The fact that we repeat the same logging 3 times is kind of noisy.

Why not add an actual error type to capture the situation of an `"Unsafe eviction policy found..."`, and let this function bubble up errors appropriately.

```
async fn check_eviction_policy(pool) -> Result<(), Error> {
    let mut conn = pool.get().await?;
    let results = redis::cmd(...)...?;
    let eviction_policy = results.get(1).ok_or_else(/* error */)?;
    if [
        "noeviction",
        "volatile-lru",
        "volatile-lfu",
        "volatile-random",
        "volatile-ttl",
    ]
    .contains(&eviction_policy.as_str()) {
       Ok(())
     } else {
       Err(...)
     }
}
```

And then the callers can just do `check_eviction_policy.inspect_err(|e| tracing::error!("{e:?}")`, or however you want to handle the errors.